text/0477-policy-validation.md

```diff
@@ -0,0 +1,561 @@
+* **API Bar Raiser**: @iliapolo
+
+Allows users to validate that their CDK application complies with infrastructure policies at synthesis time. Integrates
+with third party policy-as-code tools, such as CloudFormation Guard, OPA and Checkov.
+
+## Working Backwards
+
+### Policy Validation
+
+It is possible to use policy as code tools such
+as [CloudFormation Guard](https://docs.aws.amazon.com/cfn-guard/latest/ug/what-is-guard.html)
+or [OPA](https://www.openpolicyagent.org/) to evaluate the compliance of CDK applications.
+
+Policy as code tools are integrated with CDK through a plugin mechanism. In order to add policy enforcement for a
+specific tool you need to first specify the plugin to use. You can do this by importing the plugin and applying it to a
+stage. For example, to add a CloudFormation Guard validation to a stage you would do the following:
+
+```ts
+import { CfnGuardValidator } from '@aws-cdk/cfn-guard-validator';
+
+// globally for the entire app (an app is a stage)
+const app = new App({
+  validationPlugins: [
+    new CfnGuardValidator({
+      rules: [
+        // rules can be specified inline
+        Rules.fromAsset('../my-local-rules'),
+
+        // or from a remote location
+        Rules.fromDownload('https://somelocation.com/company-rules', {
+          auth: {},
+        }),
+
+        // or from an S3 location
+        Rules.fromS3(s3Location),
+
+        // or by calling an external endpoint
+        Rules.fromRequest('POST', 'https://someendpoint.com/validate'),
+      ],
+      options: {
+        someOption: "value",
+      },
+    }),
+  ],
+});
+
+// only apply to a particular stage
+const prodStage = new Stage(app, 'ProdStage', {
+  validationPlugins: [...],
+});
+```
+
+The details of how to configure the plugin will be specific to each plugin. For example, the `CfnGuardValidator` shown
+above has a `rules` property which is an array of `Rules` objects. The `Rules` class is a helper class that can be used
+to specify where the rules are located.
+
+CDK Policy Validation should never be used as a deployment gate or as an enforcement layer. It should instead be
+thought of as developer enablement. Ultimately, the development environment is under the control of individual 
+developers and development teams. It's up to them to opt in to this feature, and to ensure that the policy 
+validation is applied. The goal of CDK Policy Validation is to minimize the amount of set up needed during development, and make it as easy
+as possible. For example, a typical workflow could be:
+
+> A note on the term "deployment gate" above. It is perfectly acceptable to have
+> policy at synth run as part of you CI or CD pipeline and prevent deployments.
+> What we want to ensure is that users do not think of that "deployment gate" as
+> a way to _ensure_ that their infrastructure deployment is compliant. It should
+> instead be thought of as a way to "fail fast" and report issues before
+> deployment.
```

Comment:
I know what you're trying to say, but I would say it differently. How about something a long the lines of:

*The validation performed by the CDK at synth time can be bypassed by developers, and can therefore not be relied on as the sole mechanism of validation in large organizations. Some other mechanism to validate the same rule set more authoritatively should be set up independently, like CloudFormation Hooks or AWS Config. Nevertheless, CDK's ability to evaluate the rule set during development is still useful as it will improve detection speed and developer productivity.*

Reply:
ooh I like this better